internal/service/probe.go

```diff
@@ -28,21 +28,27 @@ var instance *ProbeService
 // ProbeService is an implementation
 // of the interface controller.ProbeService
 type ProbeService struct {
-	client *http.Client
-	probes map[string]*model.Probe
+	client            *http.Client
+	persistenceClient *persistence.PersistenceClient
 }
 
 // NewProbeService allow to create a new ProbeService
 // implemented as a singleton.
-func NewProbeService(client *http.Client) *ProbeService {
+func NewProbeService(client *http.Client) (*ProbeService, error) {
```

Comment:
No need to return an `error` here. If the ProbeService cannot be instanciated, there is no reason to let the application start. (because nothing is gonna work)
Just log an fatal error here.

Reply:
ok I will change this